### PR TITLE
Improve commit selection for 'Find file' feature

### PR DESCRIPTION
**Pull request by pmiossec**
before launching the search by
* selecting the last selected when multiple commits are selected (otherwise we don't know on which commit we search)
* selecting HEAD if the selected commit is artificial

### Before

We can't launch the feature, start searching and discover that we end up nowhere
original_find_file

### After

If on artificial commit, it switch to HEAD before searching
improve_find_file

When multiselect, only the commit we are searching in stay selected so that we know which one is really used:
multiselect_find_file

## Test methodology 

- Manual

## Test environment(s) 

- Git Extensions 33.33.33
- Build 9855deffe1093125ee7f6b1980fc88c667dc6ab8
- Git 2.40.0.windows.1
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.15
- DPI 96dpi (no scaling)
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -1960,6 +1960,14 @@ private void AddNotes()
 
         private void FindFileInSelectedCommit()
         {
+            var selectedRevisions = RevisionGrid.GetSelectedRevisions();
+            if (selectedRevisions.Count > 1 || (selectedRevisions.Count == 1 && selectedRevisions[0].IsArtificial))
+            {
+                GitRevision potentialRevision = selectedRevisions[0];
+                ObjectId targetCommit = potentialRevision.IsArtificial ? RevisionGrid.CurrentCheckout : potentialRevision.ObjectId;
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormBrowse.cs:1967]: Will not work if artificial commit is in the multiselect
Not the most important use case though, but removing the artificial commits from selectedRevisions is not that hard...

**Review by gerhardol — COMMENTED** [review]
+1

File: GitUI/CommandsDialogs/FormBrowse.cs (hunk 1 above)
Comment (pmiossec) [reply GitUI/CommandsDialogs/FormBrowse.cs:1967]: I will test again to be sure I didn't broke something when refactoring but I have tested this multi-commits with the 1st artificial and it was working...

Comment (gerhardol) [reply GitUI/CommandsDialogs/FormBrowse.cs:1967]: First artificial should work, but you could select two artificial commits

Comment (pmiossec) [reply GitUI/CommandsDialogs/FormBrowse.cs:1967]: For me this case is handled by the ternary in the if clause...

Comment (mstv) [reply GitUI/CommandsDialogs/FormBrowse.cs:1967]: I agree, Philippe. What have me missed, Gerhard?

The repetition can be avoided (and an empty selection be handled, too):
```
GitRevision? potentialRevision = selectedRevisions.FirstOrDefault();
ObjectId? targetCommit = potentialRevision?.IsArtificial is not false ? RevisionGrid.CurrentCheckout : selectedRevisions.Count == 1 ? null : potentialRevision!.ObjectId;
if (targetCommit is not null)
```

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 2]
```diff
@@ -1960,6 +1960,14 @@ private void AddNotes()
 
         private void FindFileInSelectedCommit()
         {
+            var selectedRevisions = RevisionGrid.GetSelectedRevisions();
```
Comment (mstv) [inline GitUI/CommandsDialogs/FormBrowse.cs:1963]: nit: in order to emphasize that it is not an enumerator
```suggestion
            IReadOnlyList<GitRevision> selectedRevisions = RevisionGrid.GetSelectedRevisions();
```

File: GitUI/CommandsDialogs/FormBrowse.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/CommandsDialogs/FormBrowse.cs:1967]: Yes this works.
Just null annotation (is OK for SetSelectedRevision())

```suggestion
                ObjectId? targetCommit = potentialRevision.IsArtificial ? RevisionGrid.CurrentCheckout : potentialRevision.ObjectId;
```

**Review by gerhardol — APPROVED** [review]

File: GitUI/CommandsDialogs/FormBrowse.cs (hunk 1 above)
Comment (pmiossec) [reply GitUI/CommandsDialogs/FormBrowse.cs]: Sorry @mstv I find the LoC too difficult to understand...

**Review by RussKie — APPROVED** [review]

**Review by mstv — APPROVED** [review]

**Pull request merged**